nextflow-training/exercise3/hello_examples/HelloGoodbye/HelloGoodbye.nf

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env nextflow
+
+////////////////////////////////////////////////////////////////////////////////////////////////
+// launch locally
+//   first move to directory:
+//      cd /root/bcc2020-training/nextflow-training/exercise3/hello_examples/HelloGoodbye
+// run workflow:
+//   nextflow run HelloGoodbye.nf
+////////////////////////////////////////////////////////////////////////////////////////////////
+
+myName = file(params.myName)
+greeting = file(params.greeting)
+
+process hello_world {
+    input:
+    file myName from myName
+
+    output:
+    file "Hello.txt" into hello_ch
+
+    """
+    echo Hello World! > Hello.txt
+    cat ${myName} >> Hello.txt
+    """
+}
+
+ process goodbye_world {
+     input:
+     file greeting from greeting
```

Comment:
Since HelloGoodbye is the multi-task workflow example, its supposed to take in the output of the hello_world part and use it as the input of goodbye_world. So the input here should be hello.txt or if there's a way to dynamically set that in the .config that'd be best. Basically the overall workflow should only take in one input: name.txt.

Reply:
Seems like its possible based on what the docs say about channels, the example seems to be using output of one as input of other

https://www.nextflow.io/docs/latest/basic.html#processes-and-channels